specs/commandline/key.md

```diff
@@ -0,0 +1,132 @@
+# notation key
+
+## Description
+
+Use ```notation key``` command to manage keys used for signing. User can add/update/list/remove key to/from signing key list. Please be noted this command doesn't manage the lifecycle of signing key itself, it manages the signing key list only.
+
+## Outline
+
+### notation key command
+
+```text
+Manage keys used for signing
+
+Usage:
+  notation key [command]
+
+Available Commands:
+  add         Add key to signing key list
+  list        List keys used for signing
+  delete      Delete key from signing key list
+  update      Update key in signing key list
+
+Flags:
+  -h, --help  help for key
+```
+
+### notation key add
+
+```text
+Add key to signing key list
+
+Usage:
+  notation key add [flags] <key_name> [key_path cert_path]
+
+Flags:
+  -d, --default                mark as default
+  -h, --help                   help for add
+      --id string              key id (required if --plugin is set)
+  -p, --plugin string          signing plugin name
+      --plugin-config strings  {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values
+```
+
+### notation key list
+
+```text
+List keys used for signing
+
+Usage:
+  notation key [flags] list
+
+Aliases:
+  list, ls
+
+Flags:
+  -h, --help                   help for list
+
+```
+
+### notation key delete
+
+```text
+Delete key from signing key list
+
+Usage:
+  notation key delete [flags] <key_name>...
+
+Aliases:
+  delete, rm
+
+Flags:
+  -h, --help                   help for delete
+
+```
+
+### notation key update
+
+```text
+Update key in signing key list
+
+Usage:
+  notation key update [flags] <key_name>
```

Comment:
Whats the purpose/ intended use of update command? Is it just to update default key?

Reply:
Currently only one flag '--default', there could be more later.